lib/ansible/playbook/play.py

```diff
@@ -97,19 +98,38 @@ def __init__(self):
     def __repr__(self):
         return self.get_name()
 
+    def _validate_hosts(self, attribute, name, value):
+        # Only validate 'hosts' if a value was passed in to original data set.
+        if 'hosts' in self._ds:
+            # Check for empty data
+            if value in ([], tuple(), set(), {}, ''):
```

Comment:
why not `if not value` ?

Reply:
I wanted the error message for `False`, `0`, and `None` to be different. Maybe that's splitting hairs too much.

Reply:
I will simply this test.